sdk/src/main/kotlin/io/rover/sdk/streams/Operators.kt

```diff
@@ -95,6 +105,61 @@ internal fun <T, R> Publisher<T>.map(transform: (T) -> R): Publisher<R> {
     }
 }
 
+/**
+ * Retry operator that when given a number of retries will not emit onError calls further downstream but will instead
+ * resubscribe to the publisher that it was previously subscribed to until the attempted number of retries is equal to the
+ * given number of retries, at this point, another onError call from upstream will be emitted downstream.
+ */
+
+internal fun <T> Publisher<T>.retry(numberOfRetries: Int): Publisher<T> {
+    val prior = this
+
+    return Publisher { subscriber ->
+        var attemptedRetries = 0
+        var subscrip: Subscription? = null
+        prior.subscribe(
+            object : Subscriber<T> {
+                override fun onComplete() {
+                    log.w("Retrying retry on complete $attemptedRetries")
+                    subscriber.onComplete()
+                }
+
+                override fun onSubscribe(subscription: Subscription) {
+                    subscrip = subscription
+                    val consumerSubscription = object : Subscription {
+                        override fun cancel() {
+                            subscription.cancel()
+                        }
+
+                        override fun request(n: Long) {
+                            if (n != Long.MAX_VALUE) throw RuntimeException("Backpressure signalling not supported.  Request Long.MAX_VALUE.")
+                            subscription.request(Long.MAX_VALUE)
+                        }
+                    }
+                    subscriber.onSubscribe(consumerSubscription)
+                }
+
+                override fun onNext(t: T) {
+                    log.w("Retrying retry on next $attemptedRetries $t")
+                    subscriber.onNext(t)
+                }
+
+                override fun onError(t: Throwable) {
+                    if (attemptedRetries < numberOfRetries) {
+                        log.w("Retrying retry $attemptedRetries ${t.debugExplanation()}")
+                        attemptedRetries++
+                        subscrip?.cancel()
+                        prior.subscribe(this)
+                    } else {
+                        subscrip?.cancel()
+                        subscriber.onError(t)
+                        log.w("Stopped Retrying retry")
```

Comment:
maybe put a `#` sign lol :)

Reply:
I think i'll just delete these and only bring them back for when i'm debugging/testing